packages/template-retail-react-app/app/pages/account/wishlist/index.jsx

```diff
@@ -21,69 +21,88 @@ import WishlistPrimaryAction from './partials/wishlist-primary-action'
     const {formatMessage} = useIntl()
     const toast = useToast()
+
     const [selectedItem, setSelectedItem] = useState(undefined)
-    const [localQuantity, setLocalQuantity] = useState({})
     const [isWishlistItemLoading, setWishlistItemLoading] = useState(false)
-    const wishlist = useWishlist()
+
+    const {data: wishListData, isLoading: isWishListLoading} = useWishList()
+    const productIds = wishListData?.customerProductListItems?.map((item) => item.productId)
+
+    const {data: productsData, isLoading: isProductsLoading} = useProducts(
+        {parameters: {ids: productIds?.join(','), allImages: true}},
+        {enabled: Boolean(productIds)}
+    )
+
+    const wishListItems = wishListData?.customerProductListItems?.map((item, i) => {
+        return {
+            ...item,
+            product: productsData?.data?.[i]
+        }
+    })
+
+    const updateCustomerProductListItem = useShopperCustomersMutation(
+        'updateCustomerProductListItem'
+    )
+    const {data: customer} = useCurrentCustomer()
 
     const handleActionClicked = (itemId) => {
         setWishlistItemLoading(!!itemId)
         setSelectedItem(itemId)
     }
 
     const handleItemQuantityChanged = async (quantity, item) => {
-        // This local state allows the dropdown to show the desired quantity
-        // while the API call to update it is happening.
-        setLocalQuantity({...localQuantity, [item.productId]: quantity})
         setWishlistItemLoading(true)
         setSelectedItem(item.productId)
-        try {
-            await wishlist.updateListItem({
-                ...item,
-                quantity: parseInt(quantity)
-            })
-        } catch {
-            toast({
-                title: formatMessage(API_ERROR_MESSAGE),
-                status: 'error'
-            })
+
+        const body = {
+            ...item,
+            quantity: parseInt(quantity)
         }
-        setWishlistItemLoading(false)
-        setSelectedItem(undefined)
-        setLocalQuantity({...localQuantity, [item.productId]: undefined})
-    }
+        // To meet expected schema, remove the custom `product` we added
+        // TODO: better way?
+        delete body.product
 
-    useEffect(() => {
-        if (customer.isRegistered) {
-            // We want to reset the wishlist here
-            // because it is possible that a user
-            // adds an item to the wishlist on another page
-            // and the wishlist page may not have enough
-            // data to render the page.
-            // Reset the wishlist will make sure the
-            // initialization state is correct.
-            if (wishlist.isInitialized) {
-                wishlist.reset()
+        updateCustomerProductListItem.mutate(
+            {
+                body,
+                parameters: {
+                    customerId: customer.customerId,
+                    itemId: item.id,
+                    listId: wishListData.id
```

Comment:
🤔 this `wishListData` can be undefined because it's from a query.
```suggestion
                    listId: wishListData?.id
```